internal/assert/assert.go

```diff
@@ -0,0 +1,70 @@
+package assert
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const errf = "unknown type for %s: %T (value: %v)"
+
+func Int(v any, name string) (int, error) {
+	switch t := v.(type) {
+	case nil:
+		return 0, nil
+	case int:
+		return t, nil
+	case int64:
+		return int(t), nil
+	case float32:
+		return int(t), nil
+	case float64:
+		return int(t), nil
+	case string:
+		return strconv.Atoi(t)
+	}
+
+	return 0, fmt.Errorf(errf, name, v, v)
+}
+
+func Int64(v any, name string) (int64, error) {
```

Comment:
If you made NumberOrString generic on the return type you wouldn't need this function at all.

```golang
type ConvertibleNumber interface { int | int64 | float32 | float64 }
```